### PR TITLE
Add STIS notebook calstis_2d_ccd.ipynb

### PR DESCRIPTION
**Pull request by dulude**
**Relevant Ticket**
- [SPB-1346](https://jira.stsci.edu/browse/SPB-1346)

**Summary**
- Added notebook _calstis_2d_ccd.ipynb_ (migrated from [spacetelescope/STIS-Notebooks@7e45ca6](https://github.com/spacetelescope/STIS-Notebooks/tree/7e45ca6802e4d216307a133a5c825117d385c499)) and associated files
- All previous commit history has been preserved in the migration of this notebook from the [STIS-Notebooks](https://github.com/spacetelescope/STIS-Notebooks) repository to [hst_notebooks](https://github.com/spacetelescope/hst_notebooks) repository
- Created notebook-specific requirements.txt file
- Updated _toc.yml accordingly

### CODE REVIEW

**Comment by haticekaratay** [comment]
Hi @Jackie-Brown, It appears that the absence of necessary files for calibration causes the assertion error. To resolve this issue, you will need to set the environment variable for the `oref` as demonstrated on this page: https://stistools.readthedocs.io/en/latest/gettingstarted.html. However, this is just my initial assumption based on the information I found in the documentation. I recommend contacting one of the scientists or authors to confirm if this solution is correct. Thank you.

File: notebooks/STIS/calstis/calstis_2d_ccd.ipynb [hunk 1]
```diff
@@ -0,0 +1,729 @@
+{
```
[the remark this answers is not in the document]
Comment (Jackie-Brown) [reply notebooks/STIS/calstis/calstis_2d_ccd.ipynb:1]: <p>Added "Flowchart of CCD data reduction and processing steps common to imaging and spectroscopy." Let me know if more detail is needed.</p>

[the remark this answers is not in the document]
Comment (Jackie-Brown) [reply notebooks/STIS/calstis/calstis_2d_ccd.ipynb:1]: <p>Added &lt;img alt="Graph illustrating parallel serial overscan corresponding to wavelength in the x-axis and virtual overscan corresponding to positions along slit in the y-axis."&gt;</p>

[the remark this answers is not in the document]
Comment (haticekaratay) [reply notebooks/STIS/calstis/calstis_2d_ccd.ipynb:1]: Thanks for addressing my comments. Looks good to me!

**Pull request merged**